staging/src/k8s.io/client-go/tools/leaderelection/leaderelection.go

```diff
@@ -326,6 +335,11 @@ func (le *LeaderElector) tryAcquireOrRenew(ctx context.Context) bool {
 	// 1. obtain or create the ElectionRecord
 	oldLeaderElectionRecord, oldLeaderElectionRawRecord, err := le.config.Lock.Get(ctx)
 	if err != nil {
+		// if TolerateNetworkPartition the Leader keeps the lease until the network is ready
+		if le.IsLeader() && le.config.TolerateNetworkPartition && errors.IsNetworkError(err) {
```

Comment:
Does this make a binary from network partition locally maintain a lease forever?  This doesn't seem like a good idea.

Reply:
it does not maintain the lease forever, it just doesn't consider that it has lost the lease, it keeps working with the previous state, once it recovers the connectivity it will "fight" against for the lease ...
I think that this is the same behavior implemented in the kubelet, when it looses the connection with the apiserver, pods keeps running, the goal here is to give the users an option to deal with the connectivity problem by themselves and not be affected by the `OnStoppedLeadingcallback` hook